torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -199,7 +199,11 @@ def runtime_wrapper(args: List[Any]):
             for idx in indices_of_inps_to_detach:
                 if isinstance(args_[idx], torch.Tensor):
                     args_[idx] = args_[idx].detach()
-            with torch.autograd._force_original_view_tracking(True):
+            # We need enable_grad when user runs compiled region under no_grad,
+            # and inputs have requires_grad. Subregion of fn can be under enable_grad.
```

Comment:
This comment doesn't quite explain what's going on:

(1) we know that if we `traced_joint == True`, we **must** be in the training case (and have some gradients to compute)

(2) It is possible to be in the `traced_joint == True` case even when the global grad state is disabled, via a user's `no_grad()` region (how? there can be a nested `enable_grad()` inside the function that forces some outputs to require gradients)


(3) therefore, whenever `trace_joint == True`, we need to unconditionally flip the grad state back on while we are executing the compiled function.